apps/analysis/serializers.py

```diff
@@ -0,0 +1,106 @@
+import json
+import tempfile
+from typing import IO, Any
+
+import pandas as pd
+from django.core.serializers.json import DjangoJSONEncoder
+from django.utils import timezone
+
+from apps.analysis.models import Resource, ResourceMetadata
+
+
+def create_resource_for_data(team, data: Any, name: str) -> Resource:
+    serializer = get_serializer(data)
+    metadata = serializer.get_metadata(data)
+    time_suffix = timezone.now().replace(tzinfo=None).isoformat(timespec="milliseconds")
+    resource = Resource(
+        team=team,
+        name=f"{name} {time_suffix}",
+        type=metadata.format,
+        metadata=metadata.model_dump(),
+    )
+    with tempfile.TemporaryFile(mode="w+") as file:
```

Comment:
Should this not also be saved on S3?

Reply:
Right! For some reason I thought that only saved a reference to the file. Which is technically true, but I forgot that it actually saves the file as well. Cool cool